src/addIndex.js

```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```

Comment:
I would welcome a clearer version of this signature.  But the function has some inherent complexity in that it's supposed to handle various kinds of cases, including:

**Map:**

    (inVal -> outVal) -> list -> result
    ((inVal, idx, list) -> outVal) -> list -> result

**Reduce**

    ((accumVal, currVal) -> newVal) -> initVal -> list -> result
    ((accumVal, currVal, idx, list) -> newVal) -> initVal -> list -> result

The nested ellipses are meant to represent the difference between `inVal -> outVal` and `(accumVal, currVal) -> newVal`, allowing for the possibility that a function might have still more parameters than this.  The top-level ellipses are to represent the difference between `fn -> list -> result` and `fn -> initVal -> list -> result`, again with the possibility that there might be even more parameters.

Any suggestions would be welcome.